pennylane/wires.py

```diff
@@ -352,11 +363,17 @@ def shared_wires(list_of_wires):
                     "Expected a Wires object; got {} of type {}.".format(wires, type(wires))
                 )
 
+        first_wires_obj = list_of_wires[0]
+        sets_of_wires = [wire.toset() for wire in list_of_wires]
+        # This find the intersection of the labels of all wires in O(n) time.
+        intersecting_wires = functools.reduce(
+            lambda a, b: a & b, sets_of_wires, first_wires_obj.toset()
+        )
         shared = []
         # only need to iterate through the first object,
         # since any wire not in this object will also not be shared
-        for wire in list_of_wires[0]:
-            if all(wire in wires_ for wires_ in list_of_wires):
+        for wire in first_wires_obj.tolist():
+            if wire in intersecting_wires:
```

Comment:
Most likely yes.

Reply:
Actually since this is in `shared_wires`, I didn't really see extra overhead in the mini benchmark I ran. (However, what you said would basically apply to the overhead seen in `all_wires`).